pkg/cmd/stack/deploy.go

```diff
@@ -500,7 +503,12 @@ func deployDeployment(ctx context.Context, svcName string, s *model.Stack, c kub
 		if isNewDeployment {
 			return false, fmt.Errorf("error creating deployment of service '%s': %s", svcName, err.Error())
 		}
-		return false, fmt.Errorf("error updating deployment of service '%s': %s", svcName, err.Error())
+		if err := deployments.Destroy(ctx, old.Name, old.Namespace, c); err != nil {
+			return false, fmt.Errorf("error updating deployment of service '%s': %s", svcName, err.Error())
+		}
+		if _, err := deployments.Deploy(ctx, d, c); err != nil {
+			return false, fmt.Errorf("error updating deployment of service '%s': %s", svcName, err.Error())
+		}
```

Comment:
This is odd, why if it failed because a temporal error redeploying? Any error in a redeploy would imply to destroy a deploy again even if it is not necessary, right?

Reply:
you are right, in case of statefulsets we make this only when the error is not "Forbidden", do you think we should control other types of errorS?

Reply:
@ifbyol i've reverted the change here and added an exception prior to deploying for first time to check if is the exception, if yes, then we destroy and redeploy

Reply:
ok, I think now looks good.

Maybe we can include a comment before the if to explain why this piece of code is needed? If not, in a few months we won't remember why we do that check

Reply:
i've added a comment with a link to this PR and some context